pyproject.toml

```diff
@@ -45,7 +45,7 @@ dev = [
     "psutil>=5.9.3,<6",
     "pydocstyle>=6.3.0,<7",
     "pydoctor>=24.11.1,<25",
-    "pyright==1.1.402",
+    "pyright==1.1.403",
```

Comment:
Asked similarly at https://github.com/nexus-rpc/sdk-python/pull/13#discussion_r2204980623, should we fixate to a specific Pyright version? Will this ensure we miss new things that may effect our users in later versions? I see we use a fixed mypy version too. I don't necessarily mind it so long as we keep on top of updating regularly.